examples/llama.py

```diff
@@ -160,7 +160,7 @@ def build(model_path, tokenizer_path, model_gen="1", model_size="7B", quantize=F
     model = Transformer(**params["args"], linear=AbsmaxQuantizedLinear, max_context=MAX_CONTEXT, jit=jit) if quantize else Transformer(**params["args"], max_context=MAX_CONTEXT, jit=jit)
 
     if model_path.is_dir():
-      weights = concat_weights([load(filename) for filename in [f"{model_path}/consolidated.{i:02d}.pth" for i in range(params["files"])]])
+      weights = concat_weights([load(filename) for filename in [f"{model_path}/consolidated.{i:02d}.pth" for i in range(params["files"])]], device[0] if isinstance(device, tuple) else device)
```

Comment:
Does this work because somewhere earlier in __main__ that all devices are inited? Can we have a loop somewhere that does `Device[device] for device in devices` to explicitly init

Reply:
I believe the cause of the child process leakage is on line 109, because when the child process spawns, it loads the function definitions.  The use of the default argument Device.DEFAULT means each child will init and not tear down the GPU resources (like AQLQueue) even if it doesn't call that function.

Reply:
oh got it, that makes sense